drivers/tee/optee/core.c

```diff
@@ -27,6 +36,102 @@
 
 #define OPTEE_SHM_NUM_PRIV_PAGES	CONFIG_OPTEE_SHM_NUM_PRIV_PAGES
 
+#if defined(CONFIG_OPTEE_HV_IKGT)
+#define OPTEE_VMCALL_SMC       0x6F707400
+#endif
+
+#if defined(CONFIG_OPTEE_VSOCK)
+#define VIRTIO_SHM_COPY_REQ 	0x5a5a5a5a
+#define VIRTIO_SMC_BUFFER_LEN	0x40
+#define VIRTIO_VSOCK_BUFF_LEN	0x10000
+#define VIRTIO_VSOCK_TEE_PORT	1234
+
+static struct socket *host_smc_sock = NULL;
+static struct socket *tee_sock = NULL;
+unsigned long optee_shm_offset = 0;
+
+static DEFINE_SEMAPHORE(optee_smc_lock);
+#endif
+
+#if defined(CONFIG_OPTEE_IVSHMEM)
+#define PCI_DRV_NAME	"optee-ivshmem"
+
+#define IVPOSITION_OFF	0x08
+#define DOORBELL_OFF	0x0C
+
+#define OPTEE_SHM_QUEUE_SIZE	64
+
+#define OPTEE_HANDLE_DONE		0xa5a5a5a5
+
+#define OPTEE_SHM_SMC_SIZE		0x200000
+
+struct ivshmem_private {
+	struct pci_dev *dev;
+
+	u8                  revision;
+	u32                 ivposition;
+
+	u8 __iomem          *regs_addr;
+	u8                  *base_addr;
+	u32                 *msix_addr;
+
+	unsigned long       bar0_addr;
+	unsigned long       bar0_len;
+	unsigned long       bar1_addr;
+	unsigned long       bar1_len;
+	unsigned long       bar2_addr;
+	unsigned long       bar2_len;
+
+	char                (*msix_names)[256];
+	struct msix_entry   *msix_entries;
+	int                 nvectors;
+};
+
+static struct ivshmem_private g_ivshmem_dev;
+
+static struct pci_device_id ivshmem_id_table[] = {
+    { 0x1af4, 0x1110, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0},
+    { 0 },
+};
+MODULE_DEVICE_TABLE(pci, ivshmem_id_table);
+
+struct optee_smc_args {
+	u64 a0;
+	u64 a1;
+	u64 a2;
+	u64 a3;
+	u64 a4;
+	u64 a5;
+	u64 a6;
+	u64 a7;
+	u64 a8;
+} __packed;
+
+struct optee_vm_ids {
+	u32 ree_id;
+	u32 tee_id;
+} __packed;
+
+struct optee_smc_ring {
+	u16 head;
+	u16 tail;
+	u16 ring[OPTEE_SHM_QUEUE_SIZE];
+} __packed;
+
+static struct optee_smc_args *g_smc_args = NULL;
+static struct optee_smc_ring *smc_avail_ring = NULL;
```

Comment:
In line 90 and 121, global variable for file scope contains prefix "g_", suggest to keep same style for line 122/123/124